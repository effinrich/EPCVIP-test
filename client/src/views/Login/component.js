/** @format */
import React, { PureComponent } from 'react'
import styled from 'styled-components'
import LoginForm from 'grommet/components/LoginForm'
import { Redirect } from 'react-router-dom'
import PropTypes from 'prop-types'
import { space } from 'styled-system'
import { Helmet } from 'react-helmet'
import { SubmissionError } from 'redux-form/immutable'

import theme from 'theme'
import Flex from 'components/Flex'
import ShadowBox from 'components/ShadowBox'
import Heading from 'components/Text/Heading'
import BodyCopy from 'components/Text/BodyCopy'
import FlatButton from 'components/FlatButton'
import apiErrorHandler from 'utils/apiErrorHandler'

// import LoginForm from './partials/LoginForm'

const StyledLogin = styled.div`
  ${space};
`

export default class LoginComponent extends PureComponent {
  static propTypes = {
    token: PropTypes.string,
    loginUser: PropTypes.func.isRequired,
    error: PropTypes.string
  }

  handleSubmit = async data => {
    const { error: loginError, payload } = await this.props.loginUser({
      email: data.get('email').toLowerCase(),
      password: data.get('password')
    })

    if (loginError) {
      throw new SubmissionError({
        _error: apiErrorHandler(payload, payload.error)
      })
    }
  }

  render() {
    const { token } = this.props

    if (token) {
      return <Redirect to='/dashboard' />
    }

    return (
      <StyledLogin pt={[2, 4]}>
        <Helmet>
          <title>Login</title>
        </Helmet>
        <Flex justifyContent='center'>
          <ShadowBox level={1} bg='white' p={3} width={[9 / 10, 1 / 2]}>
            <Heading fontSize={4}>Welcome back</Heading>
            <BodyCopy pb={2}>
              Sign in to manage your Freebird Business Account
            </BodyCopy>
            <LoginForm onSubmit={this.handleSubmit} />
            <BodyCopy pt={3} textAlign='center'>
              Not a business and need support? Please click the green help
              button in the bottom right of the screen.
            </BodyCopy>
            <FlatButton
              overBg='none'
              overColor={theme.primaryButtonOverBg}
              to='/password-recovery'
              width={1}
              color={theme.brandColor}
            >
              Forgot password?
            </FlatButton>
          </ShadowBox>
        </Flex>
      </StyledLogin>
    )
  }
}
