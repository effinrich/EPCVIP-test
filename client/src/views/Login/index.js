/** @format */
import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { Redirect } from 'react-router-dom'
import { useFormik } from 'formik'
import * as yup from 'yup'
import styled from 'styled-components'
import { space } from 'styled-system'
import { Box, FormField, TextInput, Button } from 'grommet'

import ErrorBox from 'components/ErrorBox'
import { login } from 'store/auth'

const StyledLogin = styled(Box)`
  ${space};
`

//setting the initial values
const initialValues = {
  email: '',
  password: ''
}

const validationSchema = yup.object({
  email: yup
    .string('Enter your email')
    .email('Enter a valid email')
    .required('Email is required'),
  password: yup.string('Enter your password').required('Password is required')
})

const Login = () => {
  const dispatch = useDispatch()
  const { auth, error: fetchError } = useSelector(state => state.auth)

  const formik = useFormik({
    initialValues,
    validationSchema,
    onSubmit: async (values, { setSubmitting }) => {
      dispatch(login(values))
      setSubmitting()
    }
  })

  const emailProps = formik.getFieldProps('email')
  const passwordProps = formik.getFieldProps('password')

  if (auth) {
    return <Redirect to="/" />
  }

  return (
    <StyledLogin fill align="center" justify="center">
      <Box width="medium" pad="medium" round="xsmall" background="white">
        <form onSubmit={formik.handleSubmit}>
          <FormField label="Email" error={formik.errors.email}>
            <TextInput
              name="email"
              type="email"
              a11yTitle="Email input field"
              {...emailProps}
            />
          </FormField>
          <FormField label="Password" error={formik.errors.email}>
            <TextInput
              name="password"
              type="password"
              a11yTitle="Password input field"
              {...passwordProps}
            />
          </FormField>
          <Box margin={{ top: 'medium' }} direction="row">
            <Button
              fill="horizontal"
              type="submit"
              primary
              label="Login"
              // disabled={!(formik.isValid && formik.dirty)}
            />
          </Box>
        </form>
        {fetchError && <ErrorBox>{fetchError}</ErrorBox>}
      </Box>
    </StyledLogin>
  )
}

export default Login
