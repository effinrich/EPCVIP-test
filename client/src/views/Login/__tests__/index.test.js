import React from 'react'
import { render /*, screen*/, cleanup } from 'utils/custom-test-render'

import Login from '../index'

afterEach(cleanup)

describe('Login View Component', () => {
  it('renders without crashing', () => {
    const { asFragment } = render(<Login />)
    expect(asFragment()).toMatchSnapshot()
  })
})
