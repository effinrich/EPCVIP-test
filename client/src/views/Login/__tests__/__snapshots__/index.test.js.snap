// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login View Component renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="StyledBox-sc-13pk1d4-0 bxevCE sc-bdfBwQ FAtxg"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 fMrygi"
    >
      <form>
        <div
          class="StyledBox-sc-13pk1d4-0 dLXLoO FormField__FormFieldBox-m9hood-0 jGEMXj"
        >
          <label
            class="StyledText-sc-1sadyjn-0 drTEXR"
          >
            Email
          </label>
          <div
            class="StyledBox-sc-13pk1d4-0 gmReJc FormField__FormFieldContentBox-m9hood-1"
          >
            <div
              class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
            >
              <input
                aria-label="Email input field"
                autocomplete="off"
                class="StyledTextInput-sc-1x30a0s-0 kfWKkd"
                name="email"
                placeholder="Enter your email"
                type="email"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 dLXLoO FormField__FormFieldBox-m9hood-0 jGEMXj"
        >
          <label
            class="StyledText-sc-1sadyjn-0 drTEXR"
          >
            Password
          </label>
          <div
            class="StyledBox-sc-13pk1d4-0 gmReJc FormField__FormFieldContentBox-m9hood-1"
          >
            <div
              class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
            >
              <input
                aria-label="Password input field"
                autocomplete="off"
                class="StyledTextInput-sc-1x30a0s-0 kfWKkd"
                name="password"
                placeholder="Enter your password"
                type="password"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 cOxGhq"
        >
          <button
            class="StyledButton-sc-323bzc-0 eMUCep"
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
